src/key.cpp

```diff
@@ -74,23 +73,28 @@ CPubKey CKey::GetPubKey() const {
     return result;
 }
 
+extern "C"
```

Comment:
I had the same thought process and agree with your conclusion. I don't the the external C linkage is necessary, though?